src/Algebra/Module/Properties.agda

```diff
@@ -0,0 +1,97 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of algebraic structures defined over some other
+-- structure, like modules and vector spaces
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Module.Properties where
+
+open import Algebra
+open import Algebra.Module
+open import Level
+
+private
+  variable
+    c ℓ : Level
+
+self-leftSemimodule : (R : Semiring c ℓ) → LeftSemimodule R c ℓ
+self-leftSemimodule semiring = record
+  { Carrierᴹ = Carrier
+  ; _≈ᴹ_ = _≈_
+  ; _+ᴹ_ = _+_
+  ; _*ₗ_ = _*_
+  ; 0ᴹ = 0#
```

Comment:
Would it make sense to leave all these out, since Agda can infer them?

Reply:
The reason I haven't done this is because I think seeing these fields being filled in is useful. This definition is pretty subtle, and though it's all inferable by Agda, I don't think it's inferable by humans who don't already know the theory. Furthermore, Agda can only infer these fields after lots of little lemmas are filled in, which makes me believe that it's slightly brittle.